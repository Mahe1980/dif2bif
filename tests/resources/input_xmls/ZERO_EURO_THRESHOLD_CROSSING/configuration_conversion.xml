<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration verbose="false" debugMode="false">
    <generalConfiguration>
        <properties>
            <property name="hdfsRootPath" value="/data/raw"/>
            <property name="localRootPath" value="/var/SP/data/staging"/>
        </properties>
    </generalConfiguration>
    <dataConversions>
		<dataConversion configurationId="3fc8e5b7-be0c-489e-a62c-e3dcf601673b" description="ZERO_EURO_THRESHOLD_CROSSING">
            <properties>
                <property name="country" value="it"></property>
                <property name="datasource" value="meg"></property>
                <property name="interface" value="ZERO_EURO_THRESHOLD_CROSSING"></property>
                <property name="version" value="1.0"></property>
                <property name="contains_header" value="true"></property>
                <property name="DeleteInputDataAfterSuccessfulConversion" value="false"></property>
            </properties>
            <inputFileSelectionCriteria>
                <criteria name="MinimumNumberOfFiles" value="1"></criteria>
            </inputFileSelectionCriteria>
            <dataRoot path="${hdfsRootPath}/vf_${country}/${datasource}/${interface}/${version}"></dataRoot>
            <conversionActions>
                <readAction name="ReadFromCsvAction">
                    <files type="delimited" separator="|"/>
                </readAction>
                <conversionAction name="BasicConversionAction"></conversionAction>
                <writeAction name="WriteToParquetAction">
                    <hive>
                        <database name="raw_it"></database>
                        <tableName name="zero_euro_threshold_crossing"></tableName>
                    </hive>
                </writeAction>
            </conversionActions>
            <schema>
                <field name="TAG" type="string" required="false"/>
                <field name="MSISDN" type="string" required="false"/>
                <field name="GENERATION_TIMESTAMP" type="string" required="false"/>
                <field name="BALANCE" type="string" required="false"/>
            </schema>
            <partitions>
                <partition name="year" fieldType="integer" position="0:0" source="filepath:year" />
                <partition name="month" fieldType="integer" position="0:0" source="filepath:month" />
                <partition name="day" fieldType="integer" position="0:0" source="filepath:day" />
                <partition name="hour" fieldType="integer" position="0:0" source="filepath:hour" />
            </partitions>
            <fieldActions>
                <fieldAction fieldName="MSISDN" actionType="function" actionParameter="anonymise" actionMethod="bdpmanonitmsisdn"/>
            </fieldActions>
        </dataConversion>
 </dataConversions>
</configuration>

