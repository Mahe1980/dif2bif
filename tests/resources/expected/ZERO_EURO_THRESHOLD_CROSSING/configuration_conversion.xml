<configuration verbose="false" debugMode="false">
  <generalConfiguration>
    <properties>
      <property name="rootPath" value="gs://vfit-dh-rawingested"/>
      <property name="gcsOutputRootPath" value="gs://vfit-dh-meg-rawprepared"/>
    </properties>
  </generalConfiguration>
  <dataConversions>
    <dataConversion configurationId="3fc8e5b7-be0c-489e-a62c-e3dcf601673b" description="ZERO_EURO_THRESHOLD_CROSSING">
      <properties>
        <property name="country" value="it"/>
        <property name="datasource" value="meg"/>
        <property name="interface" value="ZERO_EURO_THRESHOLD_CROSSING"/>
        <property name="version" value="1.0"/>
        <property name="DeleteInputDataAfterSuccessfulConversion" value="false"/>
        <property name="bigQueryProject" value="dh"/>
        <property name="bigQueryDataCategory" value="meg"/>
      </properties>
      <inputFileSelectionCriteria>
        <criteria name="MinimumNumberOfFiles" value="1"/>
      </inputFileSelectionCriteria>
      <dataRoot path="${rootPath}/${datasource}/${interface}/${version}"/>
      <conversionActions>
        <readAction name="ReadFromCsvAction">
          <files type="delimited" separator="|" contains_header="true" charset="utf-8"/>
        </readAction>
        <conversionAction name="BasicConversionAction"/>
        <writeActions>
          <writeAction name="WriteToBigQueryAction">
            <config>
              <dataset name="vf${country}_${bigQueryProject}_lake_${bigQueryDataCategory}_rawprepared_s"/>
              <tableName name="zero_euro_threshold_crossing"/>
            </config>
          </writeAction>
          <writeAction name="WriteToParquetAction">
            <config>
              <path>${gcsOutputRootPath}/${interface}/${version}/parquet</path>
            </config>
          </writeAction>
        </writeActions>
      </conversionActions>
      <schema>
        <field name="TAG" type="string" required="false"/>
        <field name="MSISDN" type="string" required="false"/>
        <field name="GENERATION_TIMESTAMP" type="string" required="false"/>
        <field name="BALANCE" type="string" required="false"/>
      </schema>
      <partitions>
        <partition name="fileRegex" source="property" position="(IT_BAL_MEG_H_ZERO_THRESH_CROSS)_(\d{4})(\d{2})(\d{2})(\d{2})(.*).csv.gz"/>
        <partition name="2" source="property" position="year"/>
        <partition name="3" source="property" position="month"/>
        <partition name="4" source="property" position="day"/>
        <partition name="5" source="property" position="hour"/>
      </partitions>
      <fieldActions>
        <fieldAction fieldName="MSISDN" actionType="function" actionParameter="anonymise" actionMethod="bdpmanon${anon_version}msisdn"/>
      </fieldActions>
    </dataConversion>
  </dataConversions>
</configuration>
